packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,568 @@
+    return new _RenderSegmentedControl<T>(
+      textDirection: Directionality.of(context),
+      selectedValue: selectedValue,
+      pressedValue: pressedValue,
+      keys: keys,
+    );
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, _RenderSegmentedControl<T> renderObject) {
+    renderObject
+      ..textDirection = Directionality.of(context)
+      ..selectedValue = selectedValue
+      ..pressedValue = pressedValue;
+  }
+}
+
+class _SegmentedControlContainerBoxParentData extends ContainerBoxParentData<RenderBox> {}
+
+class _RenderSegmentedControl<T> extends RenderBox
+    with ContainerRenderObjectMixin<RenderBox, ContainerBoxParentData<RenderBox>> {
+  _RenderSegmentedControl({
+    List<RenderBox> children,
+    @required T selectedValue,
+    @required T pressedValue,
+    @required TextDirection textDirection,
+    @required this.keys,
+  })  : _textDirection = textDirection,
+        _selectedValue = selectedValue,
+        _pressedValue = pressedValue {
+    addAll(children);
+  }
+
+  T get selectedValue => _selectedValue;
+  T _selectedValue;
+  set selectedValue(T value) {
+    if (_selectedValue == value) {
+      return;
+    }
+    _selectedValue = value;
+    markNeedsPaint();
+  }
+
+  T get pressedValue => _pressedValue;
+  T _pressedValue;
+  set pressedValue(T value) {
+    if (_pressedValue == value) {
+      return;
+    }
+    _pressedValue = value;
+    markNeedsPaint();
+  }
+
+  TextDirection get textDirection => _textDirection;
+  TextDirection _textDirection;
+  set textDirection(TextDirection value) {
+    if (_textDirection == value) {
+      return;
+    }
+    _textDirection = value;
+    markNeedsLayout();
+  }
+
+  List<T> keys;
+
+  List<RRect> get rects => _rects;
+  final List<RRect> _rects = <RRect>[];
+
+  List<Paint> get fillPaint => _fillPaint;
+  final List<Paint> _fillPaint = <Paint>[];
+
+  static Size _boxSize(RenderBox box) => box == null ? Size.zero : box.size;
+
+  static Rect _boxRect(RenderBox box) {
+    final _SegmentedControlContainerBoxParentData childData = box.parentData;
+    return box == null ? Rect.zero : childData.offset & box.size;
+  }
+
+  static double _minWidth(RenderBox box, double height) {
+    return box == null ? 0.0 : box.getMinIntrinsicWidth(height);
+  }
+
+  static double _minHeight(RenderBox box, double width) {
+    return box == null ? 0.0 : box.getMinIntrinsicHeight(width);
+  }
+
+  @override
+  double computeMinIntrinsicWidth(double height) {
+    double total = 0.0;
+    RenderBox child = firstChild;
+    while (child != null) {
+      final _SegmentedControlContainerBoxParentData childParentData = child.parentData;
+      total += _minWidth(child, height);
+      child = childParentData.nextSibling;
+    }
+    return total;
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) => computeMinIntrinsicWidth(height);
+
+  @override
+  double computeMinIntrinsicHeight(double width) => _minHeight(firstChild, width);
```

Comment:
i would have expected the segmented control's minimum intrinsic height to be the max of the min heights of all the widgets